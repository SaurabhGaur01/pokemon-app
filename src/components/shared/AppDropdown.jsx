import React from 'react';
import PropTypes from 'prop-types';

const AppDropdown = ({
    label,
    options,
    onSelection,
}) => {
    return (
        <div className="input-group input-group-sm mb-3">
            <div className="input-group-prepend">
                <span className="input-group-text" id="inputGroup-sizing-sm">
                    {label}
                </span>
            </div>
            <select
                className="form-select"
                aria-label="Default select example"
                onChange={onSelection}
                data-testid='connectedDropdown'
            >
                {
                    options.map(({ value, text }, index) => (
                        <option key={index} value={value}>{text}</option>
                    ))
                }
            </select>
        </div>
    );
}

AppDropdown.propTypes = {
    label: PropTypes.string.isRequired,
    options: PropTypes.array.isRequired,
    onSelection: PropTypes.func.isRequired,
};

export default AppDropdown;