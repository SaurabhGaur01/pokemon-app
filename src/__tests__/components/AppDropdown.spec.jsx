import React from 'react';
import { shallow } from 'enzyme';
import AppDropdown from '../../components/shared/AppDropdown';

let mockProps;

describe('AppDropdown', () => {
    beforeEach(() => {
        mockProps = {
            label: 'mockLabel',
            options: [{ value: 'mockValue', text: 'mockText' }],
            onSelection: jest.fn(),
        }
    })
    it('should render component correctly', () => {
        const renderedModule = shallow(<AppDropdown {...mockProps} />);
        expect(renderedModule).toMatchSnapshot();
    });

    it('should call onSelection when any dropdown value changed', () => {
        const renderedModule = shallow(<AppDropdown {...mockProps} />);
        renderedModule.find('#form-select').at(0).simulate('change', {target: { value: 'mockValue'}});
        expect(mockProps.onSelection).toHaveBeenCalledTimes(1);
    });
})