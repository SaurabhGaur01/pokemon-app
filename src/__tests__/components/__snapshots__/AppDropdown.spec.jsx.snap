// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDropdown should render component correctly 1`] = `
<div
  className="input-group input-group-sm mb-3"
>
  <div
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
      id="inputGroup-sizing-sm"
    >
      mockLabel
    </span>
  </div>
  <select
    aria-label="Default select example"
    className="form-select"
    data-testid="connectedDropdown"
    id="form-select"
    onChange={[MockFunction]}
  >
    <option
      key="0"
      value="mockValue"
    >
      mockText
    </option>
  </select>
</div>
`;
